pkg/signature/verification.go

```diff
@@ -0,0 +1,60 @@
+package verification
+
+import "errors"
+
+type (
+	SignatureVerifyAlgorithm func(pubKey string, content string, signature string) bool
+	SignatureScheme          string
+
+	Verifier interface {
+		//TODO use domain.Signature
+		Verify(algorithm SignatureScheme, pubKey string, content string, signature string) (bool, error)
+	}
+	verifier struct {
+		verifyAlgorithms map[SignatureScheme]SignatureVerifyAlgorithm
+	}
+)
+
+const (
+	Ed25519   SignatureScheme = "ed25519"
+	Secp256k1 SignatureScheme = "secp256k1"
+	Sr25519   SignatureScheme = "sr25519"
+)
+
+var ErrVerifierNotSupportedAlgorithm = errors.New("signature verifier doesn't support current algorithm")
+
+func CreateVerifier(algorithms ...SignatureScheme) (Verifier, error) {
```

Comment:
probably it make sense to rename it to 'CreateScheme' where scheme can have methods to 'sign' and 'verify'